drivers/disk/flashdisk.c

```diff
@@ -405,6 +406,8 @@ static int disk_flash_access_ioctl(struct disk_info *disk, uint8_t cmd, void *bu
 		*(uint32_t *)buff = ctx->page_size / ctx->sector_size;
 		k_mutex_unlock(&ctx->lock);
 		return 0;
+	case DISK_IOCTL_CTRL_INIT:
+		return disk_flash_access_init(disk);
```

Comment:
Shouldn't we follow this with a hard requirement that all operations on disk should return error on uninitialized device, unless the `DISK_IOCTL_CTRL_INIT` is issued?
Shouldn't `DISK_IOCTL_CTRL_DEINIT` make devices refuse to do anything unless init is issued again?

We can also go soft on requirement and just state that init is always required before we do anything an deinit will require additional init, with no check in code. That will require less code, but can it cause some strange errors?

Reply:
Lets go with the soft requirement. 
If that keeps biting us, we can change to configurable/hard requirements, to allow detection of problems.
But I guess that most such problems would be with things in a tree anyway, so on us.

The changes, if we choose to do them, would be limited to https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/disk/disk_access.c anyway, so no backends should require change if we do that later.

Hard requirement adds code and, I assume, will rarely trigger any error.

Do you agree on such assumptions?

Reply:
Yes, I do- let's make this a soft requirement